backend/app/crud/evaluations/batch.py

```
@@ -0,0 +1,202 @@
+
+    Raises:
+        Exception: If any step fails
+    """
+    try:
+        # Step 1: Fetch dataset items from Langfuse
+        logger.info(f"Starting evaluation batch for run '{eval_run.run_name}'")
+        dataset_items = fetch_dataset_items(
+            langfuse=langfuse, dataset_name=eval_run.dataset_name
+        )
+
+        if not jsonl_data:
```

Ruff (0.14.3)

Undefined name `jsonl_data`

(F821)